momaland/utils/parallel_wrappers.py

```diff
@@ -0,0 +1,90 @@
+"""Various wrappers for Parallel MO environments."""
+
+import numpy as np
+from gymnasium.wrappers.normalize import RunningMeanStd
+from pettingzoo.utils.wrappers.base_parallel import BaseParallelWrapper
+
+
+class LinearizeReward(BaseParallelWrapper):
+    """Convert MO reward vector into scalar SO reward value.
+
+    `weights` represents the weights of each objective in the reward vector space.
+    """
+
+    def __init__(self, env, weights: dict):
+        """Reward linearization class initializer.
+
+        Args:
+            env: base env to add the wrapper on.
+            weights: a dict where keys are agents and values are vectors representing the weights of their rewards.
+        """
+        self.weights = weights
+        super().__init__(env)
+
+    def step(self, actions):
+        """Returns a reward scalar from the reward vector."""
+        observations, rewards, terminations, truncations, infos = super().step(
+            actions
+        )  # super.step is called to have env.agents reachable, otherwise main loop never ends
```

Comment:
comment is outdated?

Reply:
no, this problem is still here...

Reply:
ah i understand i should have made the documentation clearer.

`agents` is used in `env.step()` to know which agents are terminated so that they are dropped from the array. **However**, this `agents` array is not in the actual `env`, it is in `BaseParallelWrapper`, which is what our env becomes after we wrap it...

I know this is confusing and I'm also struggling with it. When you look at `BaseParallelWrapper.step()` and `env.step()` the problem becomes apparent with the usage of `env.agents`. I think a solution might be to remove Line-

oh i see now... you fixed it in the PR 🤡

yes this is outdated. I was running into this issue because i had my local fix for `__getattr()__` and not your PR fix on my machine 🤡